docs/src/main/sphinx/functions/lambda.rst

```diff
@@ -15,7 +15,102 @@ Lambda expressions are written with ``->``::
     x -> CAST(x AS JSON)
     x -> x + TRY(1 / 0)
 
+Examples
+--------
+
+Obtain the squared elements of an array column::
+
+    SELECT numbers,
+           transform(numbers, n -> n * n) as squared_numbers
+    FROM (
+        VALUES
+            (ARRAY[1, 2]),
+            (ARRAY[3, 4]),
+            (ARRAY[5, 6, 7])
+    ) AS t(numbers);
+
+.. code-block:: text
+
+      numbers  | squared_numbers
+    -----------+-----------------
+     [1, 2]    | [1, 4]
+     [3, 4]    | [9, 16]
+     [5, 6, 7] | [25, 36, 49]
+    (3 rows)
+
+Besides the values of the array column being manipulated,
+other columns can be captured as well within the lambda expression.
+The following example provides a showcase of this feature
+for calculating the value of ``f(x) = ax + b`` linear function::
+
+    SELECT xvalues,
+           a,
+           b,
+           transform(xvalues, x -> a * x + b) as linear_function_values
+    FROM (
+        VALUES
+            (ARRAY[1,2], 10, 5),
+            (ARRAY[3,4], 4, 2)
+    ) AS t(xvalues, a, b);
+
+.. code-block:: text
+
+     xvalues | a  | b | linear_function_values
+    ---------+----+---+------------------------
+     [1, 2]  | 10 | 5 | [15, 25]
+     [3, 4]  |  4 | 2 | [14, 18]
+    (2 rows)
+
+
+Find the array elements containing at least one value greater than ``100``::
+
+    SELECT numbers
+    FROM (
+        VALUES
+            (ARRAY[1,NULL,3]),
+            (ARRAY[10,20,30]),
+            (ARRAY[100,200,300])
+    ) AS t(numbers)
+    WHERE any_match(numbers, n ->  COALESCE(n,0) > 100);
+    -- [100,200,300]
+
+
+Lambda functions can be also applied in aggregation functions.
+Below is a sample the overly complex calculation of the sum of all elements of a column::
+
+    SELECT reduce_agg(value, 0, (a, b) -> a + b, (a, b) -> a + b) sum_values
+    FROM (
+        VALUES
+            (1),
+            (2),
+            (3),
+            (4),
+            (5)
+    ) AS t(value);
+    -- 15
+
+
+Limitations
+-----------
+
 Most SQL expressions can be used in a lambda body, with a few exceptions:
 
 * Subqueries are not supported: ``x -> 2 + (SELECT 3)``
 * Aggregations are not supported: ``x -> max(y)``
+
+
+
+See also
+--------
+
+Below is a curated list of Trino functions which make use of lambda expressions:
```

Comment:
Also add those links in the context above as requested.

Reply:
what "curated" means here?

is it to mean that other functions which make use of lambda expressions are not recommended for general audience?

Reply:
I think I'll just remove the line because it is misleading.
I'll just leave the references to the functions which I've identified to use lambda expressions.
Thank you for the hint @findepi

Reply:
Will the list be kept up to date? how can we enforce that?
maybe it's equally useful for the users to have some example functions that use lambda expressions here and just call this "example .." so it doesn;t look like the list is supposed to be complete

Reply:
I guess _Refererences_ as suggested by @mosabua  corresponds well for the title of this section.
I pretty much assume that the list is not going to stay too long up-to-date, but having multiple function links here is good as a starting point for curious users going through the documentation.

Reply:
I agree with not attempting to be complete in the list. Maybe we should just add that as a sentence:

The following lists contains example functions that support lambda usage:

Reply:
I am also okay with removing the list btw.

Reply:
I'll just remove it - it doesn't bring that much added value to the docs.